docs/DesignDocs/enclave_openssl_support.md

```diff
@@ -0,0 +1,90 @@
+  - It is not a pervasively used crypto library, which increases the costs of porting existing code into the enclave.
+  - Although it has the appropriate Known Answer Tests and self-tests, it has never been officially FIPS 140-2 certified.
+  - It represents a specific set of functionality that may not include what is needed by the developer, for example, Chinese national crypto algorithms.
+
+- While there is a desire for a long-term solution to enable OE SDK users to replace the crypto library used by the OE enclave runtime with an arbitrary crypto library that meets their needs, it is acknowledged that supporting the OpenSSL library inside the OE enclave runtime will address some of these needs in the meantime.
+  - OpenSSL is a widely used crypto library and is the _de facto_ standard crypto library available on many Linux distros. It is much more likely that existing code is using OpenSSL, which can be more easily ported into an enclave app.
+    - Its API surface is often used as a standard that other crypto libraries either adopt (e.g. LibreSSL, BoringSSL) or already provide adaptor shims for (e.g. WolfSSL). As a result, supporting OpenSSL in the OE runtime also paves the way for a subset of other crypto libraries to be used in OE.
+  - OpenSSL is commonly FIPS-certified as part of the Linux distros.
+    - Even though the FIPS 140-2 certification applies to specific instances of the OpenSSL binaries, the presence of a prior certification for the library often influences the decision to adopt particular crypto libraries.
+  - OpenSSL supports a different and slightly broader range of functionality and options compared to mbedTLS.
+
+## User Experience
+
+![oecrypto integration with openssl](images/oe_crypto_openssl.svg)
+
+The OE SDK package will be expanded to include:
+
+- A version of the OpenSSL libraries adapted for use in an enclave (the `openssl` component in the diagram).
+- An OE-specific wrapper implementation to allow `oeenclave` to use OpenSSL instead of mbedTLS (the `oecryptoopenssl` component in the diagram).
+
+An enclave app developer will have the choice at build time for their app to link either the `oecryptombed` or `oecryptoopenssl` wrappers against `oeenclave`, which will decide which crypto library `oeenclave` uses internally.
+
+A developer can link both `mbedcrypto` and `openssl` components into their enclave app at the same time, even if `oeenclave` can only use a single wrapper at a time. For example, an app developer may choose to retain the dependency `oeenclave` has on `oecryptombed` but also additionally link in `openssl`. Doing so would allow the enclave app to use some functionality available in OpenSSL but not mbedTLS, even though `oeenclave` continues to use mbedTLS. This would not be a recommended use case because of the increase in the Trusted Computing Base (TCB) of the enclave app, but it is allowed.
```

Comment:
Today there are the following mbedtls libraries:
- libmbedcrypto.a
- libmbedtls.a
- liboecryptombed.a
- libmbedx509.a

Would linking against liboecryptoopenssl, still give users attested-tls?

Reply:
The equivalent set of libraries for OpenSSL would be:
- liboecryptoopenssl.a
- libopensslcrypto.a (i.e. OpenSSL's libcrypto)
- libopenssltls.a (i.e OpenSSL's libssl)
(see next paragraph)

The attested TLS functionality internally uses the oecrypto abstractions and substituting the OpenSSL equivalents is required to support it.